app/controllers/foreman_ansible/concerns/ansible_roles_controller_extensions.rb

```diff
@@ -0,0 +1,62 @@
+module ForemanAnsible
+  module Concerns
+    module AnsibleRolesControllerExtensions
+      def get_variable_action(kind)
+        { 'new' => _('Add'), 'obsolete' => _('Remove'), 'update' => _('Update') }[kind]
+      end
+
+      def get_role_action(kind)
+        { 'new' => _('Import Role '), 'obsolete' => _('Remove Role'), 'old' => _('Update Role Variables') }[kind]
+      end
+
+      def get_old_roles_variables(imported_variables, role)
+        variables = { 'Add' => [], 'Remove' => [], 'Update' => [] }
+        imported_variables.each do |kind, temp_variables|
+          temp_variables.each do |temp_variable|
+            variables[get_variable_action(kind)].append(temp_variable.key) if temp_variable.ansible_role_id == role.id
+          end
+        end
+        variables
+      end
+
+      def variables_to_s(variables)
+        str = ''
+        variables.each do |action, temp_variables|
+          str += "#{action}: #{temp_variables.size} " unless temp_variables.empty?
```

Comment:
nit: You may need to add a separator when there are multiple actions so the final text could be: `Add: 4, Remove 2` instead of `Add: 4 Remove: 2`

Reply:
done